File: two_sum.java
import java.util.HashMap;

public int[] twoSum(int[] nums, int target) {
    int n = nums.length;
    for (int i = 0; i < n; i++){
        for (int j = i + 1; j < n; j++) {
            if (nums[i] + nums[j] == target){
                return new int[]{i, j};
            }
        }
    }
    return new int[]{};
}


public int[] twoSum(int[] nums, int target) {
    HashMap<Integer, Integer> numMap = new HashMap<>();
    for (int i = 0; i < nums.length; i++) {
        int complement = target - nums[i];
        if (numMap.containsKey(complement)) {
            return new int[]{numMap.get(complement), i}; // Found the pair
        }
        numMap.put(nums[i], i); // Store index of the current number
    }
    return new int[]{}; // No solution found
}

File: two_sum.py
def twoSum(nums, target):
    n = len(nums)
    indices = []
    for i in range(0, n):
        for j in range(1, n):
            if (nums[i] + nums[j]) == target:
                indices.append(i)
                indices.append(j)
                return indices
            

def twoSum(nums, target):
    num_map = {} 
    for i, num in enumerate(nums):
        complement = target - num
        if complement in num_map:
            return [num_map[complement], i]  
        num_map[num] = i